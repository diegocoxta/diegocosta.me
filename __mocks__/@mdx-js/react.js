module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
