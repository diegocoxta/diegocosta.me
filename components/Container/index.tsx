import styles from './styles.module.css';

export default function Container({ children }: React.PropsWithChildren): React.ReactElement {
  return <section className={styles.container}>{children}</section>;
}
