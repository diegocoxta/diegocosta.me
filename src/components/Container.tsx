import styled from 'styled-components';

export default styled.section`
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
`;
