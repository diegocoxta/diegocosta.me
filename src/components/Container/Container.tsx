import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.section`
  max-width: 800px;
  margin: 16px;

  @media (min-width: 760px) {
    margin: 16px auto;
  }
`;

export interface ContainerProps {
  children: React.ReactNode;
}

export default function Container(props: ContainerProps): React.ReactElement {
  return (
    <Wrapper>
      <main>{props.children}</main>
    </Wrapper>
  );
}
