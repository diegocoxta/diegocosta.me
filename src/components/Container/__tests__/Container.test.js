import React from 'react';
import { render } from '@testing-library/react';

import Container from '../Container';

describe('<Container />', () => {
  it('renders properly', () => {
    const { baseElement, queryByText } = render(<Container>John Doe</Container>);
    expect(queryByText('John Doe')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
