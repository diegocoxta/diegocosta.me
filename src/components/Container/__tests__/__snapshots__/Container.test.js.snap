// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 16px;
}

@media (min-width:760px) {
  .c0 {
    margin: 16px auto;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        John Doe
      </main>
    </section>
  </div>
</body>
`;
